### PR TITLE
Add E2E test for puppet android app

### PR DESCRIPTION
**Pull request by vkolgi**
Adds github action workflow to run tests against trigger.
Adds test case to run tests against sauce labs.
Adds necessary boilerplate code to setup test execution.

### CODE REVIEW

File: e2e_tests/test_sauce_labs.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+            pass
+
+    time.sleep(2)
+    driver.back()
+    time.sleep(2)
+    driver.back()
+
+class TestAppium(unittest.TestCase):
+    def test_tab_4(self):
+        import time
```
Comment (posix4e) [inline e2e_tests/test_sauce_labs.py:99]: I don't like mid file imports make sure to run black and import sort on this file

**Review by posix4e — CHANGES_REQUESTED** [review]
Amazing work and very close

- can you make sure the black linter runs across the whole of the python codebase?
- All these sleeps are harsh. We should come up with a plan with hopefully less sleeps. But for now it's fine.

**Comment by posix4e** [comment]
@vkola

**Comment by vkolgi** [comment]
Agree with the explicit sleeps. 
They are specifically used in the context where we come back from the android settings page to the app.
Sane approach is to bring forward the Chatter activity, than pressing back button twice.
However, activity to foreground didn't work for Real android device in Saucelabs but worked for the emulator on my local machine. So had to resort to this. With some checks in place, we should be able to avoid those explicit sleeps.

**Comment by posix4e** [comment]
Got it

**Pull request closed without merging**